File: ArrayHashing/1_TwoSum/main.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i, item in enumerate(nums):
            if target-item in dic:
                return [dic[target-item],i];
            else:
                dic[item] = i
        return dic

File: ArrayHashing/1_TwoSum/main.cpp

// Time Complextiy: O(n)*O(1)
// Space : O(n)
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        unordered_map<int, int> hashmap;
        for(int i = 0; i<nums.size(); i++){
            int n = target - nums[i];
            if(hashmap.find(n)!=hashmap.end()){
                ans.push_back(hashmap[n]);
                ans.push_back(i);
                return ans;
            }
            else
                hashmap[nums[i]] = i;
        }
        return ans;
    }
};

